src/compiler/checker.ts

```diff
@@ -22978,12 +22995,18 @@ namespace ts {
                 checkGrammarForGenerator(node);
             }
 
-            const links = getNodeLinks(node);
             const type = getTypeOfSymbol(getMergedSymbol(node.symbol));
             if (isTypeAny(type)) {
                 return type;
             }
 
+            contextuallyCheckFunctionExpressionOrObjectLiteralMethod(node, checkMode);
+
+            return type;
+        }
+
+        function contextuallyCheckFunctionExpressionOrObjectLiteralMethod(node: FunctionExpression | ArrowFunction | MethodDeclaration, checkMode?: CheckMode) {
+            const links = getNodeLinks(node);

```

Comment:
doesn't getTypeOfFunctionMethodModule muck with `links` for certain JS patterns? Is it possible to retrieve the incorrect  `links` here if `contextuallyCheckFuncti...` is called in the dynamic scope of `getTypeOfFunctionMethodOrModule` ?

Reply:
It doesn't much with so much as assign to multiple links caches iirc.

Reply:
The code below ensures no reentry via the context checked flag and all of this is verbatim extracted from `getTypeOfFunctionExpressionOrObjectLiteralMethod`.